packages/api/tests/endpoints/test-executions.js

```diff
@@ -1424,6 +1426,146 @@ test.serial('PUT /executions updates the record as expected in PostgreSQL/Elasti
   t.deepEqual(updatedPgRecord.final_payload, updatedExecution.finalPayload);
 });
 
+test.serial('PUT /executions overwrites a completed record with a running record', async (t) => {
+  const execution = fakeExecutionFactoryV2({
+    collectionId: t.context.collectionId,
+    parentArn: t.context.fakeApiExecutions[1].arn,
+    status: 'completed',
+    tasks: { fakeTask: 'fake' },
+  });
+
+  const updatedExecution = fakeExecutionFactoryV2({
+    ...omit(execution, ['collectionId']),
+    asyncOperationId: t.context.testAsyncOperation.id,
+    finalPayload: { outputPayload: randomId('outputPayload') },
+    parentArn: t.context.fakeApiExecutions[2].arn,
+    status: 'running',
+    error: null,
+  });
+
+  await request(app)
+    .post('/executions')
+    .send(execution)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  await t.context.executionPgModel.get(
+    t.context.knex,
+    {
+      arn: execution.arn,
+    }
+  );
+
+  await request(app)
+    .put(`/executions/${updatedExecution.arn}`)
+    .send(updatedExecution)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  const updatedPgRecord = await t.context.executionPgModel.get(
+    t.context.knex,
+    {
+      arn: execution.arn,
+    }
+  );
+  const updatedEsRecord = await t.context.esExecutionsClient.get(execution.arn);
+  const expectedEsRecord = {
+    ...omitBy(updatedExecution, isNull),
+    collectionId: execution.collectionId,
+    createdAt: updatedPgRecord.created_at.getTime(),
+    updatedAt: updatedPgRecord.updated_at.getTime(),
+  };
+  t.like(
+    updatedEsRecord,
+    {
+      ...expectedEsRecord,
+      timestamp: updatedEsRecord.timestamp,
+    }
+  );
+
+  const translatedExecution = await translatePostgresExecutionToApiExecution(
+    updatedPgRecord,
+    t.context.knex
+  );
+
+  t.deepEqual(translatedExecution, expectedEsRecord);
+});
+
+test.serial('PUT /executions removes execution fields when nullified fields are passed in', async (t) => {
+  const execution = fakeExecutionFactoryV2({
+    collectionId: t.context.collectionId,
+    parentArn: t.context.fakeApiExecutions[1].arn,
+    status: 'completed',
+    tasks: { fakeTask: 'fake' },
+  });
+
+  const updatedExecution = fakeExecutionFactoryV2({
+    ...execution,
+    status: 'running',
+    asyncOperationId: null,
+    collectionId: null,
+    cumulusVersion: null,
+    duration: 180.5,
```

Comment:
Should duration be null here?

Reply:
[d8bfba7](https://github.com/nasa/cumulus/pull/3186/commits/d8bfba7312a9909a872eda1be3158cbfc601a3e5)